### PR TITLE
Logging

### PR DESCRIPTION
**Pull request by jcorteza**

### CODE REVIEW

File: src/main/java/com/khoros/twitterapp/resources/HomeFeedResource.java [hunk 1]
```diff
@@ -33,11 +37,37 @@ public HomeFeedResource(Twitter factory) {
...
+        logger.debug("Configuration setup: {}", conf);
+
         try {
+
             List<Status> tweetsFeed = factory.getHomeTimeline();
             return Response.status(HttpURLConnection.HTTP_OK).entity(tweetsFeed).build();
+
         } catch (TwitterException feedException) {
+
+            logger.info("Timeline retrieval aborted. Twitter Exception thrown." );
```
Comment (andrewcchau) [inline src/main/java/com/khoros/twitterapp/resources/HomeFeedResource.java:52]: Is this the right logging level?

File: src/main/java/com/khoros/twitterapp/resources/HomeFeedResource.java [hunk 2]
```diff
@@ -33,11 +37,37 @@ public HomeFeedResource(Twitter factory) {
...
+
+            } else if (feedException.exceededRateLimitation()) {
+
+                logger.debug("Current Rate Limit: {}", feedException.getRateLimitStatus());
+                logger.debug("Exceed Rate Limitation: {}", feedException.exceededRateLimitation());
+
+            } else {
+
+                logger.debug("Twitter API Access Level: {}", feedException.getAccessLevel());
+                logger.debug("TwitterException Error Message: {}", feedException.getErrorMessage());
```
Comment (andrewcchau) [inline src/main/java/com/khoros/twitterapp/resources/HomeFeedResource.java]: Are these logs the correct logging level? In a similar vein, are they all necessary?

File: src/main/java/com/khoros/twitterapp/resources/StatusUpdateResource.java [hunk 3]
```diff
@@ -35,17 +40,50 @@ public StatusUpdateResource(Twitter mockFactory) {
 
     @POST
     public Response postStatus(@FormParam("message") String tweetText) {
+
         String statusText = tweetText.trim();
+
+        logger.info ("Attempting to post a Twitter status update.");
+        logger.debug("Twitter Configuration: {}", conf);
+        logger.debug("Status Update Text: {}", statusText);
```
Comment (andrewcchau) [inline src/main/java/com/khoros/twitterapp/resources/StatusUpdateResource.java]: You can condense some of these into just 1 log statement

File: src/main/java/com/khoros/twitterapp/resources/StatusUpdateResource.java [hunk 4]
```diff
@@ -35,17 +40,50 @@ public StatusUpdateResource(Twitter mockFactory) {
...
+
+            } else if (tweetException.exceededRateLimitation()) {
+
+                logger.info("Current Rate Limit: {}", tweetException.getRateLimitStatus());
+                logger.debug("Exceed Rate Limitation: {}", tweetException.exceededRateLimitation());
+
+            } else {
+
+                logger.info("Twitter API Access Level: {}", tweetException.getAccessLevel());
+                logger.debug("TwitterException Error Message: {}", tweetException.getErrorMessage());
```
Comment (andrewcchau) [inline src/main/java/com/khoros/twitterapp/resources/StatusUpdateResource.java]: Are these the right logging levels? Are these logs necessary?

**Comment by kev-chan** [comment]
per our conversation earlier, is the stacktrace being logged?

**Pull request merged**